### PR TITLE
PWX-27188: move all tests to the tests/basic directory

### PR DESCRIPTION
**Pull request by pureneelesh**
**What this PR does / why we need it**:

Having tests in separate directories with their own Test* function makes ginkgo create a separate binary for each directory. This patch moves most tests into tests/basic directory.

All tests now use the same TestBasic(), TestMain(), BeforeSuite() and AfterSuite() functions.

Moved any test specific code in those functions to the Describe().BeforeEach() block of those tests.

Added flags to selectively to the cleanup in AfterSuite().

Left pds tests alone since they have a substantially different BeforeSuite() block.

**Which issue(s) this PR fixes** (optional)
PWX-27188

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by lsrinivas-pure** [comment]
Hi @pureneelesh 
can you run a torped regression job with these changes 

Also Engineering efficency team is working on some changes to reduce the image size,
I think these changes comes under same line , 
@CharudathGopal can you please look at this so that the efforts are not duplicated or can this be integrated to changes for increasing torpedo efficiency

**Review by Pure-AdamuKaapan — COMMENTED** [review]
LGTM from Pure test side

File: HOW_TO_RUN.md [hunk 1]
```diff
@@ -39,11 +39,9 @@ For example:
 
 This will run the basic test against just elastic search
 
-To run all tests: ``ginkgo -v bin/*.test --  -spec-dir `pwd`/drivers/scheduler/k8s/specs``
+To run all tests: ``ginkgo -v bin/basic.test --  -spec-dir `pwd`/drivers/scheduler/k8s/specs``
 
-To run just the reboot tests: ``ginkgo -v bin/reboot.test --  -spec-dir `pwd`/drivers/scheduler/k8s/specs``
```
Comment (jainsmit) [inline HOW_TO_RUN.md:44]: Can we add example here for running single/specific test ?

**Review by jainsmit — APPROVED** [review]
LGTM !!

**Comment by pureneelesh** [comment]
> Hi @pureneelesh can you run a torped regression job with these changes

Torpedo dev job passed except for one test: ProfileOnlyDiags . That test failed without my changes as well on a recent torpedo master build. So, I am treating that failure as not related to my changes. cc: @jrivera-px 

Ran 2 more jobs for sanity check and they passed:
https://jenkins.pwx.dev.purestorage.com/view/Control%20Plane/job/Torpedo/job/tp-scale-sharedv4-svc/273/ 
https://jenkins.pwx.dev.purestorage.com/view/Control%20Plane/job/Torpedo/job/tp-basic-sharedv4-svc/786/

**Review by lsrinivas-pure — COMMENTED** [review]
Cant find sharedv4_test.go file in the PR

**Comment by pureneelesh** [comment]
> Cant find sharedv4_test.go file in the PR

@lsrinivas-pure : there was already a file with that name in the `tests/basic` directory. So I renamed that file one while moving it. It is now called `tests/basic/sharedv4_multivol_test.go`.

**Comment by lsrinivas-pure** [comment]
> > Cant find sharedv4_test.go file in the PR
> 
> @lsrinivas-pure : there was already a file with that name in the `tests/basic` directory. So I renamed that file while moving it. It is now called `tests/basic/sharedv4_multivol_test.go`.

Hi @pureneelesh ,
can you please let me know the location of this test in your PR, Im unable to find it
https://github.com/portworx/torpedo/blob/61e1be0dbf40e8e5aa31db44dfc788e25a75225a/tests/basic/sharedv4_test.go#L26

**Comment by pureneelesh** [comment]
> Hi @pureneelesh , can you please let me know the location of this test in your PR, Im unable to find it
> 
>https://github.com/portworx/torpedo/blob/61e1be0dbf40e8e5aa31db44dfc788e25a75225a/tests/basic/sharedv4_test.go#L26

@lsrinivas-pure : that file has not changed in this PR. It was already in the `tests/basic` directory and no changes were needed.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by pureneelesh** [comment]
Rebased on the latest master, resolved the conflicts in `telemetry_test.go` and squashed the fix-ups.

**Review by Rohit-PX — APPROVED** [review]
LGTM.

**Review by jrivera-px — APPROVED** [review]
Re-ran the telemetry tests locally and they all passed.
Ran 8 of 99 Specs in 1778.143 seconds
SUCCESS! -- 8 Passed | 0 Failed | 0 Pending | 91 Skipped
PASS

**Review by mborodin-px — APPROVED** [review]
LGTM autopilot test passed on your image here: https://jenkins.pwx.dev.purestorage.com/job/Autopilot/job/tp-aut-op-nextpx-resize-pvc/243/

**Comment by pureneelesh** [comment]
Thank you all. I have merged the PR since we have enough indication that this approach will work. If there are any hiccups, I will work on those post-merge.

**Pull request merged**